### DISCUSSION TITLE
Undefined array key "data"

### DISCUSSION THREAD

**Discussion by kmaphane** [Z3d0X/filament-fabricator#57] [Q&A]
Good day, 

I am receiving this error after changing the page's name and the slug. Is there anyone able to assist? Or has anyone had this error before? 

This is the error in the stack trace:

![image](https://user-images.githubusercontent.com/80934965/232807758-af64601d-41ae-4f5e-8425-4e16987825c4.png)

**Comment by kmaphane** [comment]
This happened the first time when I tried to reuse a header between multiple pages. So I just decided to create a header for each page.

**Comment by Z3d0X** [comment]
What is "data"? Please share the exception

**Comment by kmaphane** [comment]
Morning thanks for the response:

Everything was working fine until I changed the page name and the slug for pages I had already built. I was rebuilding an existing site and first created the pages as copies of the pages I had already built eg home-copy. When I was ready to use the new pages I change the slug to home. That is when I got the undefined array key data error. 

The data I'm passing through are blocks I have built within the fabricator package. 

The exception is as below:

Undefined array key "data" (View: C:\xampp\htdocs\ele-aware\vendor\z3d0x\filament-fabricator\resources\views\components\page-blocks.blade.php) (View: C:\xampp\htdocs\ele-aware\vendor\z3d0x\filament-fabricator\resources\views\components\page-blocks.blade.php) (View: C:\xampp\htdocs\ele-aware\vendor\z3d0x\filament-fabricator\resources\views\components\page-blocks.blade.php) (View: C:\xampp\htdocs\ele-aware\vendor\z3d0x\filament-fabricator\resources\views\components\page-blocks.blade.php)
C:\xampp\htdocs\ele-aware\storage\framework\views\05a0d300a389ac1e7c21357b8263e86eb9dd1219.php#29
Illuminate\View\ViewException
Undefined array key "data" (View: C:\xampp\htdocs\ele-aware\vendor\z3d0x\filament-fabricator\resources\views\components\page-blocks.blade.php) (View: C:\xampp\htdocs\ele-aware\vendor\z3d0x\filament-fabricator\resources\views\components\page-blocks.blade.php) (View: C:\xampp\htdocs\ele-aware\vendor\z3d0x\filament-fabricator\resources\views\components\page-blocks.blade.php)
C:\xampp\htdocs\ele-aware\storage\framework\views\05a0d300a389ac1e7c21357b8263e86eb9dd1219.php#29
Illuminate\View\ViewException
Undefined array key "data" (View: C:\xampp\htdocs\ele-aware\vendor\z3d0x\filament-fabricator\resources\views\components\page-blocks.blade.php) (View: C:\xampp\htdocs\ele-aware\vendor\z3d0x\filament-fabricator\resources\views\components\page-blocks.blade.php)
C:\xampp\htdocs\ele-aware\storage\framework\views\05a0d300a389ac1e7c21357b8263e86eb9dd1219.php#29
Illuminate\View\ViewException
Undefined array key "data" (View: C:\xampp\htdocs\ele-aware\vendor\z3d0x\filament-fabricator\resources\views\components\page-blocks.blade.php)
C:\xampp\htdocs\ele-aware\storage\framework\views\05a0d300a389ac1e7c21357b8263e86eb9dd1219.php#29
Illuminate\View\ViewException
Undefined array key "data"
C:\xampp\htdocs\ele-aware\storage\framework\views\05a0d300a389ac1e7c21357b8263e86eb9dd1219.php#29

**Comment by kmaphane** [comment]
View
C:\xampp\htdocs\ele-aware\vendor\z3d0x\filament-fabricator\resources\views\components\page-blocks.blade
.php
Data
errors
Illuminate\Support\ViewErrorBag {#926 ▼
  #bags: []
}
blocks
array:8 [▼
  0 => array:2 [▶
    "type" => "home.home-header"
    "data" => array:9 [▶
      "headerThreeTitleOne" => null
      "headerThreeTitleTwo" => null
      "headerThreeSubtext" => null
      "headerOneTitleOne" => null
      "headerOneTitleTwo" => null
      "headerOneSubtext" => null
      "headerTwoTitleOne" => null
      "headerTwoTitleTwo" => null
      "headerTwoSubtext" => null
    ]
  ]
  1 => array:2 [▶
    "type" => "home.home-about-us"
    "data" => array:6 [▶
      "aboutUsPictureText" => null
      "aboutUsTitleOne" => null
      "aboutUsTitleTwo" => null
      "aboutUsStrongText" => null
      "aboutUsParagraphOne" => null
      "aboutUsParagraphTwo" => null
    ]
  ]
  2 => array:2 [▶
    "type" => "home.home-services"
    "data" => array:1 [▶
      "homeServices" => []
    ]
  ]
  3 => array:2 [▶
    "type" => "home.home-certification"
    "data" => array:3 [▶
      "homeCertificationText" => null
      "homeCertificationTitleOne" => null
      "homeCertificationTitleTwo" => null
    ]
  ]
  4 => array:2 [▶
    "type" => "home.home-banner"
    "data" => array:2 [▶
      "homeBannerTitle" => null
      "homeBannerText" => null
    ]
  ]
  5 => array:2 [▶
    "type" => "home.home-achievements"
    "data" => array:1 [▶
      "homeAchievements" => []
    ]
  ]
  6 => array:1 [▶
    "type" => "home.home-donors"
  ]
  7 => array:1 [▶
    "type" => "home.home-enterprises"
  ]
]

**Comment by kmaphane** [comment]
For home.home-enterprise I am querying the following data

 public static function mutateData(array $data): array
    {
        $data = [
            'enterprises' => Enterprise::with('media', 'enterpriseCategory')->get()
        ];

        return $data;
    }

**Comment by kmaphane** [comment]
I ended up deleting these two blocks and recreating them 

6 => array:1 [▶
"type" => "home.home-donors"
]
7 => array:1 [▶
"type" => "home.home-enterprises"

Then it worked. But I used the same code for donors the code is as below:

 $data = [];

        $donors = Donor::all();

        if ($donors->count() > 0) {
            $data['donors'] = $donors->take(4)->random();
        }

        return $data;

**Comment by kmaphane** [comment]
Thanks for your response though. I hope there is something that can be done, unless it is just my environment.
